packages/extension-api/src/extension-api.d.ts

```diff
@@ -650,16 +651,77 @@ declare module '@kortex-app/api' {
     models: Array<InferenceModel>;
   };
 
+  export interface InputResponse {
+    value: string;
+  }
+
+  export interface InputWithVariableResponse extends InputResponse {
+    variables: Record<string, InputResponse>;
+  }
+
+  export interface MCPRemoteServerConfig {
+    type: 'remote';
+    index: number;
+    headers: Record<string, InputWithVariableResponse>;
+  }
+
+  export interface MCPPackageServerConfig {
+    type: 'package';
+    index: number;
+    runtimeArguments: Record<number, InputWithVariableResponse>;
+    packageArguments: Record<number, InputWithVariableResponse>;
+    environmentVariables: Record<string, InputWithVariableResponse>;
+  }
+
+  export type MCPServerConfig = MCPRemoteServerConfig | MCPPackageServerConfig;
+
+  export type MCPServerDetail = components['schemas']['ServerDetail'];
+
+  export type MCPServer = {
+    server: MCPServerDetail;
+    config: MCPServerConfig;
+  };
+
+  export type RagProviderConnection = {
+    name: string;
+    mcpServer: MCPServer;
+    credentials(): Record<string, string>;
+    index(doc: Uri, chunks: Uri[]): Promise<void>;
+    unindex(doc: Uri): Promise<void>;
```

Comment:
question: not sure about `unindex`, `deindex`

Reply:
my point was that `unindex` seems not the right word here. `deindex` or `removeIndex`, `clearIndex`